### PR TITLE
refactor: [M3-7334] - Improve, clean up, and simplify app entrypoint and render tree logic

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝
The goal of this PR is to simplify the entry point logic of Cloud Manager. Currently, the flow of `index.tsx ➡️ AuthenticationWrapper.tsx ➡️ App.tsx ➡️ MainContent.tsx ➡️ ...` is very complex and hard to follow, this ticket should help a little bit. There is a lot to clean up and improvement potential, but this PR is a solid start

## Changes  🔄
- Splashscreen simplification
- AuthenticationWrapper is now a hook
- IdentifyUser is now a hook
- Uses less redux to track app loading state
- Don't use unnecessarily complex preference toggle for Side Menu state, use preferences hooks directly

### Verification steps 
- Verify Cloud Manager loads like normal
- Log out and verify you can log in without any issues
- Verify that your remain logged in when you refresh
- Verify Lish still works

## As an Author I have considered 🤔

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/store/authentication/authentication.reducer.ts [hunk 1]
```diff
@@ -12,19 +12,21 @@ import {
 import { clearLocalStorage } from './authentication.helpers';
 import { State } from './index';
 
-export const defaultState: State = {
-  expiration: null,
-  loggedInAsCustomer: false,
-  scopes: null,
-  token: null,
-};
```
Comment (bnussman-akamai) [inline packages/manager/src/store/authentication/authentication.reducer.ts:20]: No reason to default to `null` when we can synchronously know the default values form local storage

File: packages/manager/src/hooks/useInitialRequests.ts
Comment (bnussman-akamai) [inline packages/manager/src/hooks/useInitialRequests.ts:1]: This is basically `AuthenticationWrapper` converted into a hook

File: packages/manager/src/components/AuthenticationWrapper/AuthenticationWrapper.test.tsx
Comment (bnussman-akamai) [inline packages/manager/src/components/AuthenticationWrapper/AuthenticationWrapper.test.tsx:1]: This enzyme test isn't providing much value anymore. Many of the test cases are incorrect and we have plenty of e2e coverage that checks that the app loads properly

File: packages/manager/src/hooks/useInitialRequests.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        queryFn: getUserPreferences,
+        queryKey: 'preferences',
+      }),
+    ];
+
+    // Start events polling
+    startEventsInterval(store, queryClient);
+
+    try {
+      await Promise.all(dataFetchingPromises);
```
Comment (abailly-akamai) [inline packages/manager/src/hooks/useInitialRequests.ts:94]: `Promise.all` is all or nothing - if any promise fails Promise. all() method will return an error and it will not take into account whether other promises are successfully fulfilled or not. I think this is the way we were already doing things? Not that's it's bad, just wanna make sure this is what we want. We will still load the application, but for instance (unless I am missing something here), if the preferences call fail, would be nice to still get the rest.

File: packages/manager/src/index.tsx [hunk 3]
```diff
@@ -7,17 +7,17 @@ import { ReactQueryDevtools } from 'react-query/devtools';
...
-import SplashScreen from 'src/components/SplashScreen';
+import { SplashScreen } from 'src/components/SplashScreen';
 import 'src/exceptionReporting';
 import Logout from 'src/layouts/Logout';
 import { setupInterceptors } from 'src/request';
 import { storeFactory } from 'src/store';
 
 import { App } from './App';
 import { LinodeThemeWrapper } from './LinodeThemeWrapper';
+import NullComponent from './components/NullComponent';
```
Comment (abailly-akamai) [inline packages/manager/src/index.tsx:20]: 😭 this is the component - do we really need this?
```
const NullComponent = () => {
  return null;
};

export default NullComponent;
```

**Review by abailly-akamai — COMMENTED** [review]
Awesome cleanup 🎉 . Adds a lot of clarity to our entry point. I think it's pretty close, I have not noticed any issue while testing locally but left some comments to see about improving a few things

File: packages/manager/src/components/SplashScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+
+import { CircleProgress } from 'src/components/CircleProgress';
+import { srSpeak } from 'src/utilities/accessibility';
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/components/SplashScreen.tsx:4]: > M3-7352

File: packages/manager/src/index.tsx (hunk 3 above)
Comment (bnussman-akamai) [reply packages/manager/src/index.tsx:20]: Haha, it is used by the e2e test and by the Jest config. I hate it but we have to leave it for now

File: packages/manager/src/hooks/useInitialRequests.ts (hunk 2 above)
Comment (bnussman-akamai) [reply packages/manager/src/hooks/useInitialRequests.ts:94]: Yeah, I see what you mean. It doesn't really matter what fails here, because either way, we unblock rendering and the React Query hooks may continue to fetch (depending on their retry logic). I think for the case of just showing the Splash screen, it's okay. 

I'm not really saying this is how I would implement it, I'm just trying to make incremental improvements on what we have already.

File: packages/manager/src/hooks/useInitialRequests.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+      // preferences
+      queryClient.prefetchQuery({
+        queryFn: getUserPreferences,
+        queryKey: 'preferences',
+      }),
+    ];
+
+    // Start events polling
+    startEventsInterval(store, queryClient);
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/hooks/useInitialRequests.ts:91]: This function itself should't throw. Also, events will be revamped when we React Query-ify them

File: packages/manager/src/MainContent.tsx [hunk 6]
```diff
@@ -249,7 +240,7 @@ const MainContent = (props: CombinedProps) => {
    *
    * So in this case, we'll show something more user-friendly
    */
-  if (props.globalErrors.account_unactivated) {
+  if (globalErrors.account_unactivated) {
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/MainContent.tsx:243]: `globalErrors` won't be undefined

File: packages/manager/src/MainContent.tsx [hunk 7]
```diff
@@ -283,138 +269,125 @@ const MainContent = (props: CombinedProps) => {
...
   }
 
+  const desktopMenuIsOpen = preferences?.desktop_sidebar_open ?? false;
+
+  const desktopMenuToggle = () => {
+    updatePreferences({
+      desktop_sidebar_open: !preferences?.desktop_sidebar_open,
+    });
+  };
+
```
Comment (carrillo-erik) [inline packages/manager/src/MainContent.tsx:283]: I really like this much cleaner approach.

File: packages/manager/src/store/authentication/authentication.reducer.ts [hunk 8]
```diff
@@ -12,19 +12,21 @@ import {
...
-
 const {
   expire: expiryInLocalStorage,
   scopes: scopesInLocalStorage,
   token: tokenInLocalStorage,
 } = authentication;
 
+const defaultToken = tokenInLocalStorage.get();
```
Comment (carrillo-erik) [inline packages/manager/src/store/authentication/authentication.reducer.ts:21]: Is this `const` declared here to avoid typing out `tokenInLocalStorage.get()` in line 25 and 27?

File: packages/manager/src/MainContent.tsx [hunk 9]
```diff
@@ -283,138 +269,125 @@ const MainContent = (props: CombinedProps) => {
...
+      </ComplianceUpdateProvider>
     </div>
   );
 };
 
-export default compose<CombinedProps, Props>(
-  React.memo,
-  withGlobalErrors()
-)(MainContent);
+export default React.memo(MainContent);
```
Comment (carrillo-erik) [inline packages/manager/src/MainContent.tsx:390]: Should this be converted to a named export?

File: packages/manager/src/index.tsx (hunk 3 above)
Comment (abailly-akamai) [reply packages/manager/src/index.tsx:20]: ugh

**Review by abailly-akamai — APPROVED** [review]
Thanks for addressing the feedback.

This looks ready as far as I could test it. ✅ 

Have not found issues locally and we probably want to have this sit in develop sooner than later so we can catch any potential issue before the next release.

**Comment by tyler-akamai** [comment]
- [ ] Splashscreen simplification
- [ ] AuthenticationWrapper is now a hook
- [ ] IdentifyUser is now a hook
- [ ] Uses less redux to track app loading state
- [ ] Don't use unnecessarily complex preference toggle for Side Menu state, use preferences hooks directly

File: packages/manager/src/MainContent.tsx [hunk 10]
```diff
@@ -283,138 +269,125 @@ const MainContent = (props: CombinedProps) => {
...
-                <SideMenu
-                  closeMenu={() => toggleMenu(false)}
-                  collapse={desktopMenuIsOpen || false}
-                  open={menuIsOpen}
-                />
-                <div
-                  className={cx(classes.content, {
-                    [classes.fullWidthContent]:
-                      desktopMenuIsOpen ||
-                      (desktopMenuIsOpen && desktopMenuIsOpen === true),
```
Comment (tyler-akamai) [inline packages/manager/src/MainContent.tsx:332]: hmm this line looks a little odd, good job cleaning it up

File: packages/manager/src/features/TopMenu/TopMenu.tsx [hunk 11]
```diff
@@ -18,28 +19,23 @@ import { UserMenu } from './UserMenu';
...
-  const {
-    desktopMenuToggle,
-    isLoggedInAsCustomer,
-    isSideMenuOpen,
-    openSideMenu,
-    username,
-  } = props;
+  const { desktopMenuToggle, isSideMenuOpen, openSideMenu, username } = props;
+
+  const { loggedInAsCustomer } = useAuthentication();
```
Comment (tyler-akamai) [inline packages/manager/src/features/TopMenu/TopMenu.tsx:30]: Good cleanup

File: packages/manager/src/MainContent.tsx (hunk 7 above)
Comment (tyler-akamai) [reply packages/manager/src/MainContent.tsx:283]: same 🧹

File: packages/manager/src/MainContent.tsx (hunk 9 above)
Comment (tyler-akamai) [reply packages/manager/src/MainContent.tsx:390]: I was also wondering this.

Since this is a cleanup ticket, should we do CSS Migration or should we keep it as separate ticket?

File: packages/manager/src/hooks/useInitialRequests.ts
Comment (tyler-akamai) [reply packages/manager/src/hooks/useInitialRequests.ts:1]: LGTM

File: packages/manager/src/index.tsx [hunk 12]
```diff
@@ -7,17 +7,17 @@ import { ReactQueryDevtools } from 'react-query/devtools';
...
 import { loadDevTools, shouldEnableDevTools } from './dev-tools/load';
```
Comment (tyler-akamai) [inline packages/manager/src/index.tsx:21]: This is a small thing:
Should we dynamically import dev tools to ensure they arent part of the production bundle?

**Review by tyler-akamai — APPROVED** [review]

File: packages/manager/src/MainContent.tsx (hunk 9 above)
Comment (carrillo-erik) [reply packages/manager/src/MainContent.tsx:390]: The default to named export change here is trivial in my opinion. I also believe CSS migration work should be done as a separate ticket and labeled as such. This PR is addressing a different concern.

File: packages/manager/src/store/authentication/authentication.reducer.ts (hunk 8 above)
Comment (bnussman-akamai) [reply packages/manager/src/store/authentication/authentication.reducer.ts:21]: yes!
